File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/decryption/Contract.java
package gateway.hitrontech.com.encryption.fragment.decryption;

import moe.xing.mvp_utils.BasePresenter;
import moe.xing.mvp_utils.BaseView;

public interface Contract {

  interface Presenter extends BasePresenter {

    void getPlainText(String cipherText);


    void getCipherText();
  }

  interface View extends BaseView<Presenter> {

    void setPlainText(String plainText);

    void setCipherText(String cipherText);

  }


}

File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/decryption/Contract.kt
package gateway.hitrontech.com.encryption.fragment.decryption

import moe.xing.mvp_utils.BasePresenter
import moe.xing.mvp_utils.BaseView

interface Contract {

    interface Presenter : BasePresenter {

        fun getPlainText(cipherText: String)


        fun getCipherText()
    }

    interface View : BaseView<Presenter> {

        fun setPlainText(plainText: String)

        fun setCipherText(cipherText: String)

    }


}
